import into failed when simulate network partition Between ddl owner and all other pods

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、run tpcc
2、run import into
3、simulate network partition between ddl owner and all other pods lasts for 3mins

### 2. What did you expect to see? (Required)
import into can succeed

### 3. What did you see instead (Required)
import into failed

case log:

[2025/12/15 18:08:33.909 +08:00] [INFO] [chaos.go:219] ["fault will last for"] [duration=3m0s]
[2025/12/15 18:08:33.927 +08:00] [INFO] [chaos.go:76] ["Run chaos"] [chaosId="ns=endless-ha-test-import-into-tps-7986641-1-127,kind=network-partition,name=network-partition-nekwziub,spec=&k8s.ChaosIdentifier{Namespace:\"endless-ha-test-import-into-tps-7986641-1-127\", Name:\"network-partition-nekwziub\", Spec:NetworkPartitionSpec{Duration: \"\", Direction: both, Scheduler: <nil>}, Labels:map[string]string(nil)},labels=map[string]string(nil)"] [name=network_partition] [selectors="[endless-ha-test-import-into-tps-7986641-1-127/tc-tidb-0]"] [selectorsRetainPolicy(selectors)="[endless-ha-test-import-into-tps-7986641-1-127/tc-tidb-0]"] [targetSelectors="[endless-ha-test-import-into-tps-7986641-1-127/tc-tidb-0,endless-ha-test-import-into-tps-7986641-1-127/tc-tidb-1,endless-ha-test-import-into-tps-7986641-1-127/tc-tikv-0,endless-ha-test-import-into-tps-7986641-1-127/tc-tikv-1,endless-ha-test-import-into-tps-7986641-1-127/tc-tikv-2,endless-ha-test-import-into-tps-7986641-1-127/tc-tikv-3,endless-ha-test-import-into-tps-7986641-1-127/tc-pd-0,endless-ha-test-import-into-tps-7986641-1-127/tc-pd-1,endless-ha-test-import-into-tps-7986641-1-127/tc-pd-2,endless-ha-test-import-into-tps-7986641-1-127/br-0]"] [targetSelectorsRetainPolicy(targetSelectors)="[endless-ha-test-import-into-tps-7986641-1-127/tc-tidb-0,endless-ha-test-import-into-tps-7986641-1-127/tc-tidb-1,endless-ha-test-import-into-tps-7986641-1-127/tc-tikv-0,endless-ha-test-import-into-tps-7986641-1-127/tc-tikv-1,endless-ha-test-import-into-tps-7986641-1-127/tc-tikv-2,endless-ha-test-import-into-tps-7986641-1-127/tc-tikv-3,endless-ha-test-import-into-tps-7986641-1-127/tc-pd-0,endless-ha-test-import-into-tps-7986641-1-127/tc-pd-1,endless-ha-test-import-into-tps-7986641-1-127/tc-pd-2,endless-ha-test-import-into-tps-7986641-1-127/br-0]"] [experimentSpec="NetworkPartitionSpec{Duration: \"\", Direction: both, Scheduler: <nil>}"]

[2025/12/15 18:11:33.928 +08:00] [INFO] [chaos.go:94] ["Clean chaos"] [name=network_partition] [chaosId="ns=endless-ha-test-import-into-tps-7986641-1-127,kind=network-partition,name=network-partition-nekwziub,spec=&k8s.ChaosIdentifier{Namespace:\"endless-ha-test-import-into-tps-7986641-1-127\", Name:\"network-partition-nekwziub\", Spec:NetworkPartitionSpec{Duration: \"\", Direction: both, Scheduler: <nil>}, Labels:map[string]string(nil)},labels=map[string]string(nil)"]

Test case ######ha_tidb(ddl-owner)_to_all_network_partition###### failed
the status of import job is not finished or running (now: 2025-12-15 18:12:06, jobId: 60004, step: validating, status: failed)
operatorLogs:
[2025-12-15 17:43:32] ###### start import into
import into user_data1 from 's3://qe-testing/brie/lightning-mhy-100G-csv/sysbench.user_data1.000001*.csv?access-key=xxxxxx&secret-access-key=xxxxxx&endpoint=http%3a%2f%2fks3-cn-beijing-internal.ksyuncs.com&force-path-style=false&region=Beijing&provider=ks' WITH DETACHED,thread=8,skip_rows=1
[2025-12-15 17:43:32] ###### wait for import job to finish
[2025-12-15 18:12:06] ###### wait for import job to finish failed
select id, table_schema, table_name, step, status,parameters from mysql.tidb_import_jobs where start_time >= '2025-12-15 17:43:32' and table_schema = 'importtest' and table_name = 'user_data1'
jobId: 60004, table_schema: importtest, table_name: user_data1, step: validating, status: failed

tidb logs
`[2025/12/15 18:11:38.713 +08:00] [WARN] [task_executor.go:755] ["subtask failed"] [task-id=60004] [task-key=ImportInto/60004] [error="[tikv:9001]PD server timeout: start timestamp may fall behind safe point"] [errorVerbose="[tikv:9001]PD server timeout: start timestamp may fall behind safe point\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\t/root/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20250523034308-74f78ae071ee/normalize.go:177\ngithub.com/pingcap/tidb/pkg/store/driver/error.ToTiDBErr\n\t/workspace/source/tidb/pkg/store/driver/error/error.go:119\ngithub.com/pingcap/tidb/pkg/store/copr.(*kvStore).CheckVisibility\n\t/workspace/source/tidb/pkg/store/copr/store.go:48\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIterator).Next\n\t/workspace/source/tidb/pkg/store/copr/coprocessor.go:1344\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIterator).Next\n\t/workspace/source/tidb/pkg/store/copr/coprocessor.go:1313\ngithub.com/pingcap/tidb/pkg/distsql.(*selectResult).NextRaw\n\t/workspace/source/tidb/pkg/distsql/select_result.go:518\ngithub.com/pingcap/tidb/pkg/executor.(*ChecksumTableExec).handleChecksumRequest\n\t/workspace/source/tidb/pkg/executor/checksum.go:168\ngithub.com/pingcap/tidb/pkg/executor.(*ChecksumTableExec).checksumWorker\n\t/workspace/source/tidb/pkg/executor/checksum.go:144\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1693"]
[2025/12/15 18:11:38.721 +08:00] [ERROR] [task_executor.go:384] ["run subtask failed"] [task-id=60004] [task-key=ImportInto/60004] [error="[tikv:9001]PD server timeout: start timestamp may fall behind safe point"] [errorVerbose="[tikv:9001]PD server timeout: start timestamp may fall behind safe point\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\t/root/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20250523034308-74f78ae071ee/normalize.go:177\ngithub.com/pingcap/tidb/pkg/store/driver/error.ToTiDBErr\n\t/workspace/source/tidb/pkg/store/driver/error/error.go:119\ngithub.com/pingcap/tidb/pkg/store/copr.(*kvStore).CheckVisibility\n\t/workspace/source/tidb/pkg/store/copr/store.go:48\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIterator).Next\n\t/workspace/source/tidb/pkg/store/copr/coprocessor.go:1344\ngithub.com/pingcap/tidb/pkg/store/copr.(*copIterator).Next\n\t/workspace/source/tidb/pkg/store/copr/coprocessor.go:1313\ngithub.com/pingcap/tidb/pkg/distsql.(*selectResult).NextRaw\n\t/workspace/source/tidb/pkg/distsql/select_result.go:518\ngithub.com/pingcap/tidb/pkg/executor.(*ChecksumTableExec).handleChecksumRequest\n\t/workspace/source/tidb/pkg/executor/checksum.go:168\ngithub.com/pingcap/tidb/pkg/executor.(*ChecksumTableExec).checksumWorker\n\t/workspace/source/tidb/pkg/executor/checksum.go:144\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1693"]`

### 4. What is your TiDB version? (Required)
./tidb-server -V
 Release Version: v9.0.0-beta.2.pre-894-g69fb8db
Edition: Community
Git Commit Hash: 69fb8dbc9231c1a4b5d6d44534a9ae9657007c63
Git Branch: HEAD
UTC Build Time: 2025-12-15 05:44:57
GoVersion: go1.25.5
Race Enabled: false
Check Table Before Drop: false
Store: unistore
Kernel Type: Classic
2025-12-15T17:43:23.720+0800

--- a/pkg/lightning/common/retry.go
+++ b/pkg/lightning/common/retry.go
@@ -100,12 +100,17 @@ var retryableErrorIDs = map[errors.ErrorID]struct{}{
 	errdef.ErrKVRaftProposalDropped.ID(): {},
 	// litBackendCtxMgr.Register may return the error.
 	ErrCreatePDClient.ID(): {},
-	// during checksum coprocessor will transform error into driver error in handleCopResponse using ToTiDBErr
-	// met ErrRegionUnavailable on free-tier import during checksum, others hasn't met yet
+	// during checksum coprocessor, either through ADMIN CHECKSUM TABLE or by calling
+	// distsql directly, we will transform error into driver error in handleCopResponse
+	// using ToTiDBErr. we ever met ErrRegionUnavailable on free-tier import during
+	// checksum, and meet: "[tikv:9001]PD server timeout: start timestamp may fall
+	// behind safe point" when the TiDB doing checksum and network partitioned
+	// with the cluster, others hasn't met yet
 	drivererr.ErrRegionUnavailable.ID(): {},
 	drivererr.ErrTiKVStaleCommand.ID():  {},
 	drivererr.ErrTiKVServerTimeout.ID(): {},
 	drivererr.ErrTiKVServerBusy.ID():    {},
+	drivererr.ErrPDServerTimeout.ID():   {},
 	drivererr.ErrUnknown.ID():           {},
 }
 

--- a/pkg/lightning/common/retry_test.go
+++ b/pkg/lightning/common/retry_test.go
@@ -86,12 +86,17 @@ func TestIsRetryableError(t *testing.T) {
 	require.True(t, IsRetryableError(errdef.ErrKVRaftProposalDropped.GenWithStack("test")))
 	require.False(t, IsRetryableError(errdef.ErrKVDiskFull.GenWithStack("test")))
 
-	// tidb error
-	require.True(t, IsRetryableError(drivererr.ErrRegionUnavailable))
-	require.True(t, IsRetryableError(drivererr.ErrTiKVStaleCommand))
-	require.True(t, IsRetryableError(drivererr.ErrTiKVServerTimeout))
-	require.True(t, IsRetryableError(drivererr.ErrTiKVServerBusy))
-	require.True(t, IsRetryableError(drivererr.ErrUnknown))
+	for _, err := range []error{
+		// tidb error
+		drivererr.ErrRegionUnavailable,
+		drivererr.ErrTiKVStaleCommand,
+		drivererr.ErrTiKVServerTimeout,
+		drivererr.ErrTiKVServerBusy,
+		drivererr.ErrPDServerTimeout,
+		drivererr.ErrUnknown,
+	} {
+		require.True(t, IsRetryableError(errors.Annotate(err, "failed")))
+	}
 
 	// net: connection refused
 	_, err := net.Dial("tcp", "localhost:65533")
@@ -103,33 +108,41 @@ func TestIsRetryableError(t *testing.T) {
 
 	// MySQL Errors
 	require.False(t, IsRetryableError(&mysql.MySQLError{}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrUnknown}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrLockDeadlock}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrPDServerTimeout}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrTiKVServerTimeout}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrTiKVServerBusy}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrResolveLockTimeout}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrRegionUnavailable}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrWriteConflictInTiDB}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrWriteConflict}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrInfoSchemaExpired}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrInfoSchemaChanged}))
-	require.True(t, IsRetryableError(&mysql.MySQLError{Number: tmysql.ErrTxnRetryable}))
+	for _, errNumber := range []uint16{
+		tmysql.ErrUnknown,
+		tmysql.ErrLockDeadlock,
+		tmysql.ErrPDServerTimeout,
+		tmysql.ErrTiKVServerTimeout,
+		tmysql.ErrTiKVServerBusy,
+		tmysql.ErrResolveLockTimeout,
+		tmysql.ErrRegionUnavailable,
+		tmysql.ErrWriteConflictInTiDB,
+		tmysql.ErrWriteConflict,
+		tmysql.ErrInfoSchemaExpired,
+		tmysql.ErrInfoSchemaChanged,
+		tmysql.ErrTxnRetryable,
+	} {
+		require.True(t, IsRetryableError(&mysql.MySQLError{Number: errNumber}))
+	}
 
 	// gRPC Errors
 	require.False(t, IsRetryableError(status.Error(codes.Canceled, "")))
 	require.True(t, IsRetryableError(status.Error(codes.Unknown, "region 1234 is not fully replicated")))
 	require.True(t, IsRetryableError(status.Error(codes.Unknown, "No such file or directory: while stat a file "+
 		"for size: /...../63992d9c-fbc8-4708-b963-32495b299027_32279707_325_5280_write.sst: No such file or directory")))
-	require.True(t, IsRetryableError(status.Error(codes.DeadlineExceeded, "")))
-	require.True(t, IsRetryableError(status.Error(codes.NotFound, "")))
-	require.True(t, IsRetryableError(status.Error(codes.AlreadyExists, "")))
-	require.True(t, IsRetryableError(status.Error(codes.PermissionDenied, "")))
-	require.True(t, IsRetryableError(status.Error(codes.ResourceExhausted, "")))
-	require.True(t, IsRetryableError(status.Error(codes.Aborted, "")))
-	require.True(t, IsRetryableError(status.Error(codes.OutOfRange, "")))
-	require.True(t, IsRetryableError(status.Error(codes.Unavailable, "")))
-	require.True(t, IsRetryableError(status.Error(codes.DataLoss, "")))
+	for _, code := range []codes.Code{
+		codes.DeadlineExceeded,
+		codes.NotFound,
+		codes.AlreadyExists,
+		codes.PermissionDenied,
+		codes.ResourceExhausted,
+		codes.Aborted,
+		codes.OutOfRange,
+		codes.Unavailable,
+		codes.DataLoss,
+	} {
+		require.True(t, IsRetryableError(status.Error(code, "")))
+	}
 
 	// sqlmock errors
 	require.False(t, IsRetryableError(fmt.Errorf("call to database Close was not expected")))

